mlflow/lightgbm.py

```diff
@@ -182,3 +192,111 @@ def __init__(self, lgb_model):
 
     def predict(self, dataframe):
         return self.lgb_model.predict(dataframe)
+
+
+@experimental
+def autolog(importance_types=['split']):  # pylint: disable=W0102
```

Comment:
Hm is it expensive to compute the feature importances? Maybe we could just log both split & gain feature importance for now, & simplify the API by removing this parameter.

Reply:
No, it just iterates over nodes of each tree and counts the split feature (for split) or sums the split gain (for gain). I agree with logging both. Will fix this.

How feature importance is calculated: 
https://github.com/microsoft/LightGBM/blob/edb9149c13fe5a901898db525c22b441f0db6e49/src/boosting/gbdt_model_text.cpp#L553-L580

Reply:
Wrote a simple test code to confirm this.

```python
# code
model = lgb.train(params,
                  train_set,
                  num_boost_round=2,
                  early_stopping_rounds=10,  # setting larger value than num_boost_round
                  valid_sets=[train_set, valid_set],
                  valid_names=['train', 'valid'])
print(model.best_iteration)
print(model.best_score)
```

output

```
[1]     train's multi_logloss: 0        valid's multi_logloss: 0
Training until validation scores don't improve for 10 rounds
[LightGBM] [Warning] Stopped training because there are no more leaves that meet the split requirements
[2]     train's multi_logloss: 0        valid's multi_logloss: 0
Did not meet early stopping. Best iteration is:
[1]     train's multi_logloss: 0        valid's multi_logloss: 0
1
defaultdict(<class 'collections.OrderedDict'>, {'train': OrderedDict([('multi_logloss', 0.0)]), 'valid': OrderedDict([('multi_logloss', 0.0)])})
```